test/integration/test_kubernetes_staging.py

```diff
@@ -87,64 +96,95 @@
 
 def job_config(template_str, jobs_directory):
     job_conf_template = string.Template(template_str)
-    port = os.environ.get('GALAXY_TEST_PORT')
-    assert port
-    infrastructure_url = "http://%s:%s" % (GALAXY_TEST_KUBERNETES_INFRASTRUCTURE_HOST, port)
     container_amqp_url = to_infrastructure_uri(AMQP_URL)
+    instance_id = ''.join(random.choice(string.ascii_lowercase) for i in range(8))
     job_conf_str = job_conf_template.substitute(jobs_directory=jobs_directory,
                                                 tool_directory=TOOL_DIR,
+                                                instance_id=instance_id,
                                                 k8s_config_path=integration_util.k8s_config_path(),
+                                                k8s_namespace=GALAXY_TEST_KUBERNETES_NAMESPACE,
                                                 amqp_url=AMQP_URL,
                                                 container_amqp_url=container_amqp_url,
-                                                infrastructure_url=infrastructure_url,
                                                 )
     with tempfile.NamedTemporaryFile(suffix="_kubernetes_integration_job_conf.yml", mode="w", delete=False) as job_conf:
         job_conf.write(job_conf_str)
     return job_conf.name
 
 
 @integration_util.skip_unless_kubernetes()
-@integration_util.skip_unless_fixed_port()
-class KubernetesStagingContainerIntegrationTestCase(BaseJobEnvironmentIntegrationTestCase, MulledJobTestCases):
+@integration_util.skip_unless_amqp()
+class BaseKubernetesStagingTest(BaseJobEnvironmentIntegrationTestCase, MulledJobTestCases):
+    # Test leverages $UWSGI_PORT in job code, need to set this up.
+    require_uwsgi = True
 
     def setUp(self):
-        super(KubernetesStagingContainerIntegrationTestCase, self).setUp()
+        super(BaseKubernetesStagingTest, self).setUp()
```

Comment:
Can you overwrite the dataset populator with `KubernetesDatasetPopulator` here ? That should at least print out the node state.